src/util/adb.js

```diff
@@ -347,14 +347,16 @@ export default class ADBUtils {
   }
 
   async discoverRDPUnixSocket(
-    deviceId: string, apk: string,
+    deviceId: string, apk: string, msg: string,
     {maxDiscoveryTime, retryInterval}: DiscoveryParams = {}
   ): Promise<string> {
     let rdpUnixSockets = [];
 
     const discoveryStartedAt = Date.now();
 
     while (rdpUnixSockets.length === 0) {
+      log.info(`\n\x1b[93m${msg}\x1b[39m\n`);
```

Comment:
It would be nice to also add a new test case to cover this in this group of test cases:
https://github.com/mozilla/web-ext/blob/master/tests/unit/test-util/test.adb.js#L1042

we should be able to collect the logs and make an assertion about it using a testing strategy similar to what we are using here: https://github.com/mozilla/web-ext/blob/29798cbc61e78e024911150734081c2f4ddd1be6/tests/unit/test-extension-runners/test.chromium.js#L128-L152

[This other existing test case here](https://github.com/mozilla/web-ext/blob/5492a0bdbbf210348967158aa7c80e9fb032593c/tests/unit/test-util/test.adb.js#L1096-L1130) may be helpful to figure out how you could simulate a unix socket discovery that should have looped a few times and printed the expected message in the new test case.

Take a look to the code linked above and give it a a try, but feel free to ping me in this PR if you need more pointers or help to figure out issues with creating this test (also feel free to push in this PR an incomplete test case that does still fail, that would still be helpful to me because I can more quickly look to what you are trying and how it does fail and provide more helpful pointers).

Reply:
`
it('informs the user if remote debugging is not enabled', async () => {
      const adb = getFakeADBKit({
        adbClient: {
          shell: sinon.spy(() => Promise.resolve('')),
        },
        adbkitUtil: {
          readAll: sinon.spy(() => {
            return Promise.resolve(Buffer.from(''));
          }),
        },
      });
      const adbUtils = new ADBUtils({adb});

      const maxDiscoveryTime = 50;
      const retryInterval = 10;
      const msg = (
        'Waiting for org.mozilla.firefox_mybuild Remote Debugging Server...' +
        '\nMake sure to enable "Remote Debugging via USB" ' +
        'from Settings -> Developer Tools if it is not yet enabled.'
      );

      consoleStream.stopCapturing();
      consoleStream.flushCapturedLogs();
      consoleStream.startCapturing();
      consoleStream.makeVerbose();

      const promise = adbUtils.discoverRDPUnixSocket(
        'device1', 'org.mozilla.firefox_mybuild', {
          maxDiscoveryTime,
          retryInterval,
        }
      );
      const {capturedMessages} = consoleStream;
      consoleStream.stopCapturing();

      assert.ok(capturedMessages.some(
        (message) => (
          message.match('[debug]') &&
          message.match(msg)
        )));
      await assert.isRejected(promise, WebExtError);
    });
`
@rpl 
I have tried capturing messages from log and checking them here. Does this test case fulfill the required purpose or am I making some mistakes here?

Reply:
@sonalprabhu from a quick look I think that it is a good start, it may require just some tweaks, e.g.:
- it may be better to stop capturing the messages after `await assert.isRejected(...)`, just to be sure that the `adb.discoverRDPUnixSocket(...)` async call did complete
- Nit, it may be enough to match just a fragment of the logged message (e.g. just `Make sure to enable "Remote Debugging via USB"` the part), it should be enough to cover the fix and we may not need to update the test if we do apply some small tweaks to the logged message (which would always be required if we do assert the entire message)

would you mind to update this PR so that I can see it running in the CI job? (and be able to also give this PR a try locally by just pulling your pull request as is, without having to add the test manually into the test files).

would you mind to also rebase the PR to the current main branch? the previous unrelated CI job failure has been fixed in the main branch.

Reply:
Sure, will do that.